framework/src/media/MediaWorker.cpp

```diff
@@ -45,6 +45,7 @@ void MediaWorker::startWorker()
 		if (ret != OK) {
 			medvdbg("Fail to create worker thread, return value : %d\n", ret);
 			--mRefCnt;
+			mIsRunning = false;
```

Comment:
Why do you use the variable to decide thread running?
You can create pthread without any variable and call pthread_cancel when you want to stop it.
In yours,
```
set the variable to true
-> create thread
-> check the variable always during thread is alive
-> set the variable to false when you want to stop the thread
-> check the variable and then stop
```
If you use pthread_cancel, it can work like below
```
create thread without any variable setting
-> run loop without any variable check
-> call pthread_cancel when you want to stop
```

Reply:
My comment is not for @davidfather 's change. But if this module will be changed, this PR is not needed so that I mention above here.

Reply:
@sunghan-chang In that case, how to free memory which allocated local in the thread function?

Reply:
There are several ways to do it.
1. use atexit if it has only one thread or if it can release allocated memory later
2. use SIGKILL signal and callback function
3. use task manager with exit / cancel cb

Reply:
If possible, it is better to change the thread termination process with another PR.

Reply:
after merged. i will check for pthread_cancel.